### PR TITLE
PSScriptAnalyzer linter changes

### PR DESCRIPTION
**Pull request by besessener**
`Install-Module -Name PSScriptAnalyzer -Force` is required to locally install it if missing.
PSScriptAnalyzer will act as a powershell linter on both productive and test sources.
All deviations got resolved, only one was resolved by writing a justification.

### CODE REVIEW

File: powershell/spl-functions.ps1 [hunk 1]
```diff
@@ -82,22 +83,22 @@ Function PythonInstall ([string[]]$Packages, [string[]]$TrustedHosts) {
...
         }
     }
 }
 
 # installs required tools that are needed to use scoop and python as installers
-Function Install-Basic-Tools() {
+Function Install-Basic-Toolset() {
     if (-Not (Get-Command scoop -ErrorAction SilentlyContinue)) {
         if ((New-Object Security.Principal.WindowsPrincipal([Security.Principal.WindowsIdentity]::GetCurrent())).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
             & $PSScriptRoot\install-scoop.ps1 -RunAsAdmin
```
Comment (xxthunder) [inline powershell/spl-functions.ps1]:
```suggestion
            & "$PSScriptRoot\install-scoop.ps1" -RunAsAdmin

```

Comment (xxthunder) [reply powershell/spl-functions.ps1]: Please do this everywhere to handle spaces in $PSScriptRoot

**Review by xxthunder — CHANGES_REQUESTED** [review]
🦆

**Comment by besessener** [comment]
as We saw a bug for installing dependencies, this PR is also adding a new way of installing scoop packages with Scoopfile.json.

**Pull request closed without merging**